enshrine-esdt-safe/src/from_sovereign/transfer_tokens.rs

```diff
@@ -32,58 +29,41 @@ pub trait TransferTokensModule:
 
         require!(self.not_paused(), "Cannot transfer while paused");
 
-        let (operation_hash, is_registered) =
-            self.calculate_operation_hash(hash_of_hashes.clone(), operation.clone());
+        let (op_hash, is_registered) = self.calculate_operation_hash(&hash_of_hashes, &operation);
 
         if !is_registered {
             sc_panic!("Operation is not registered");
         }
 
-        let are_tokens_registered =
-            self.verify_operation_tokens_issue_paid(operation.tokens.clone());
+        let (sov_tokens, non_sov_tokens, are_tokens_registered) =
+            self.split_payments_for_prefix_and_fee(&operation.tokens);
 
         if !are_tokens_registered {
             self.emit_transfer_failed_events(
                 &hash_of_hashes,
-                &OperationTuple {
-                    op_hash: operation_hash.clone(),
-                    operation: operation.clone(),
-                },
+                &OperationTuple { op_hash, operation },
             );
 
             return;
         }
-    }
 
-    #[endpoint]
-    fn call_token_handler_mint_endpoint(&self, operation: Operation<Self::Api>) {
         let token_handler_address = self.token_handler_address().get();
+        let multi_value_tokens: MultiValueEncoded<OperationEsdtPayment<Self::Api>> =
+            non_sov_tokens.into();
 
         self.tx()
             .to(token_handler_address)
             .typed(token_handler_proxy::TokenHandlerProxy)
-            .mint_tokens(MultiValueEncoded::from(operation.tokens.clone()))
-            .callback(<Self as TransferTokensModule>::callbacks(self).save_minted_tokens())
-            .async_call_and_exit();
-    }
+            .transfer_tokens(
+                operation.data.opt_transfer_data,
+                operation.to,
+                multi_value_tokens,
+            )
+            .multi_esdt(sov_tokens)
```

Comment:
`non_sov_tokens` should be payment and `sov_tokens` should be in arguments for `transfer_tokens`